direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java

```diff
@@ -224,18 +224,20 @@ ObserveHandle processConsumer(
 
     final OffsetCommitter<TopicPartition> offsetCommitter = createOffsetCommitter();
 
-    BiConsumer<TopicPartition, ConsumerRecord<Object, Object>> preWrite =
-        (tp, r) ->
-            offsetCommitter.register(
-                tp,
-                r.offset(),
-                1,
-                () -> {
-                  OffsetAndMetadata mtd = new OffsetAndMetadata(r.offset() + 1);
-                  if (commitToKafka) {
-                    kafkaCommitMap.put(tp, mtd);
-                  }
-                });
+    final BiConsumer<TopicPartition, ConsumerRecord<Object, Object>> preWrite =
+        (tp, r) -> {
+          final long offset = r.offset();
+          offsetCommitter.register(
+              tp,
+              r.offset(),
```

Comment:
Can we make use of the new local variable here as well?

Reply:
We can, but it's just a cosmetic change with no effect.